libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/QnAMaker/QnAMakerDialog2.cs

```diff
@@ -9,125 +9,165 @@
 
 namespace Microsoft.Bot.Builder.Dialogs.Adaptive.QnA
 {
+    /// <summary>
+    /// An adaptive dialog that supports multi-step and adaptive-learning QnA Maker services.
+    /// </summary>
+    /// <remarks>An instance of this class targets a specific QnA Maker knowledge base, determined at run-time.
+    /// It supports knowledge bases that include follow-up prompt and active learning features.</remarks>
     public class QnAMakerDialog2 : QnAMakerDialog
     {
+        /// <summary>
+        /// The declarative name for this type.
+        /// </summary>
+        /// <remarks>Used by the framework to serialize and deserialize an instance of this type to JSON.</remarks>
         [JsonProperty("$kind")]
         public const string DeclarativeType = "Microsoft.QnAMakerDialog";
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QnAMakerDialog2"/> class.
+        /// The JSON serializer uses this constructor to deserialize objects of this class.
+        /// </summary>
+        /// <param name="sourceFilePath">The source file path, for debugging. Defaults to the full path
+        /// of the source file that contains the caller.</param>
+        /// <param name="sourceLineNumber">The line number, for debugging. Defaults to the line number
+        /// in the source file at which the method is called.</param>
         [JsonConstructor]
         public QnAMakerDialog2([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
             : base(sourceFilePath, sourceLineNumber)
         {
         }
 
         /// <summary>
-        /// Gets or sets the KnowledgeBase Id of your QnA Maker KnowledgeBase.
+        /// Gets or sets the path for storing and retrieving the
+        /// QnA Maker knowledge base ID to query.
```

Comment:
No, these descriptions are not right.
We should use: "the KnowledgeBaseId or an expression which evaluates to a KnowledgeBaseId."
It's not the path.

Reply:
I've updated the comments for all of the xExpression type properties.